internal/cli/server.go

```diff
@@ -80,18 +82,41 @@ func startEpinioServer(wg *sync.WaitGroup, port int, ui *termui.UI) (*http.Serve
 	return srv, listeningPort, nil
 }
 
-// loggingmiddleware for requests
+// logging middleware for requests
 func logRequestHandler(h http.Handler, ui *termui.UI) http.Handler {
 	fn := func(w http.ResponseWriter, r *http.Request) {
+		// log the request first, then ...
+		logRequest(r, ui)
 
-		// call the original http.Handler
+		// ... call the original http.Handler
 		h.ServeHTTP(w, r)
-
-		// log the request
-		uri := r.URL.String()
-		method := r.Method
-		ui.Normal().V(1).Msgf("%s %s", method, uri)
 	}
 
 	return http.HandlerFunc(fn)
 }
+
+func logRequest(r *http.Request, ui *termui.UI) {
+	uri := r.URL.String()
+	method := r.Method
+	ui.Normal().V(1).Msgf("%s %s", method, uri)
+
+	msg := ui.Normal().V(1).Compact()
+
+	// Read request body for logging
+	bodyBytes, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		msg.Msgf("RequestBody Error: %v", err)
+		return
+	}
+	r.Body.Close()
+
+	// Recreate body for the actual handler
+	r.Body = ioutil.NopCloser(bytes.NewBuffer(bodyBytes))
+
+	if len(bodyBytes) == 0 {
+		msg.Msg("RequestBody: n/a")
+		return
+	}
+
+	msg.Msgf("RequestBody\n%s\n/RequestBody", string(bodyBytes))
```

Comment:
In some cases the request body could have sensitive data in it. For example when creating custom services with credentials to external databases and such. Logging to stdout maybe result in those credentials being written in more places than needed (e.g. local filesystem  unencrypted).

Among other problems, people may need to send out the server logs for debugging when they have issues. It's easy to leak credentials without them noticing. If you decide to keep it as is in this PR, please create an issue to find a solution to this problem. Maybe we can disable logging the request body in some actions, when we expect sensitive data as part of the request body.

Reply:
Note that these messages are limited to `--verbosity 1`. By default the server runs at level 0.
IOW the standard install will not log anything at all.
I see this as something for dev only at the moment.
Where I can bump the verbosity level in the server deployment code.

Reply:
increasing verbosity would be the first thing we would ask a user to do if they were trying to report an issue. We don't want people to accidentally leak their credentials during some debugging. It's ok for now but we need to think about a safer solution. I still think this deserves an issue in the backlog.

Reply:
Will make an issue for it.